File: theme-tests/sample.c
#include <stdbool.h>
#include <stddef.h>
#include <stdio.h>

#define LIMIT 3

// C11: macros, typedefs, struct fields, pointers, and format placeholders.
typedef struct Swatch {
    const char *label;
    const char *color;
} Swatch;

enum Mode { MODE_CALM, MODE_VIBRANT };

/** Print a bounded list of palette labels. */
static size_t render(const Swatch *swatches, size_t count, bool enabled)
{
    if (!enabled) return 0;
    size_t shown = 0;
    for (size_t index = 0; index < count && index < LIMIT; ++index) {
        printf("%zu / %s / %s\n", index, swatches[index].label, swatches[index].color);
        ++shown;
    }
    return shown;
}

int main(void)
{
    const Swatch swatches[] = {{"Mint", "#72E0B2"}, {"Blue", "#6FA8FF"}};
    const enum Mode mode = MODE_VIBRANT;
    const size_t count = sizeof swatches / sizeof swatches[0];
    return render(swatches, count, mode == MODE_VIBRANT) > 0 ? 0 : 1;
}

File: theme-tests/sample.php
<?php
declare(strict_types=1);

namespace HiAstra\Preview;

use RuntimeException;

// PHP 8.1+: variables, properties, types, attributes, and embedded HTML.
#[\Attribute]
final class Swatch
{
    public const LIMIT = 3;

    /** Build a readable label without changing the palette. */
    public function __construct(
        public readonly string $name,
        public readonly string $color = '#72E0B2',
    ) {}

    public function label(int $index): string
    {
        if ($index < 0) {
            throw new RuntimeException('Invalid index');
        }
        return sprintf('%d / %s', $index, $this->name);
    }
}

$enabled = true;
$options = ['title' => 'HiAstra', 'limit' => Swatch::LIMIT];
$swatches = [new Swatch('Mint'), new Swatch('Blue', '#6FA8FF')];
foreach ($swatches as $index => $swatch) {
    if ($enabled && $index < $options['limit']) {
        echo $swatch->label($index) . "\n";
    }
}
?>
<section class="preview"><?= htmlspecialchars($options['title']) ?></section>
